android/src/app/organicmaps/search/SearchAdapter.java

```diff
@@ -139,7 +141,10 @@ void bind(@NonNull SearchResult result, int order)
       formatOpeningHours(mResult);
       UiUtils.setTextAndHideIfEmpty(mDescription, mResult.getFormattedDescription(mFrame.getContext()));
       UiUtils.setTextAndHideIfEmpty(mRegion, mResult.description.region);
-      UiUtils.setTextAndHideIfEmpty(mDistance, mResult.description.distance.toString(mFrame.getContext()));
+      if (LocationHelper.INSTANCE.getMyPosition() != null)
+        mDistance.setText(mResult.description.distance.toString(mFrame.getContext()));
+      else
+        mDistance.setVisibility(View.GONE);
```

Comment:
This is a regression.
Before refactoring, mResult.description.distance.toString returned null or empty if there was no position.
Why now it returns zero?

Reply:
UPD: mResult.description.distance as a string was null before. Now also better to make it null or empty, but checking LocationHelper inside this Adapter class is not a good idea.

Reply:
It returned `""` because distance was just a string. Now it's a class which is initialized with zero by default.
I can think of some solutions:
1. cover it with `std::optional` which
2. keep flag `isInitialized` somewhere nearby
3. initialize it with `-1` by default and introduce `IsValid()` function

The 3rd one seems to be ok and we can try to implement it this way. What do you think?

Reply:
Why is 'std::optional' not good? Wouldn't writing the new function IsValid() not be needed?
(disclaimer: I'm trying to learn 😅)

Reply:
`std::optional` stores value in the heap. We have a lot of Distance variables, so, there will be a lot of memory allocations and deallocations which leads to bad performance.

Reply:
Maybe not in the heap, but with some additional flag, so the memory size will be 8 (double) + 8 (flag with possible alignment).

Reply:
Ah, I see! Thank you 😊 I've mostly been coding in python so far, so I hadn't even considered allocation and the heap!

Reply:
std::optional does not use heap, value + flag are stored on the stack.

Reply:
I see, so solution 3 is indeed the most efficient, as only the value (-1) is stored, and the function is called dynamically, instead of storing a flag. 
Many thanks 😊